pkg/koordlet/runtimehooks/nri/server.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nri
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/nri/pkg/api"
+	"github.com/containerd/nri/pkg/stub"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/resourceexecutor"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/hooks"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/runtimehooks/protocol"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+	rmconfig "github.com/koordinator-sh/koordinator/pkg/runtimeproxy/config"
+)
+
+type nriconfig struct {
+	Events []string `json:"events"`
+}
+
+type Options struct {
+	NriSocketPath string
+	// support stop running other hooks once someone failed
+	PluginFailurePolicy rmconfig.FailurePolicyType
+	// todo: add support for disable stages
+	DisableStages map[string]struct{}
+	Executor      resourceexecutor.ResourceUpdateExecutor
+}
+
+type NriServer struct {
+	cfg             nriconfig
+	stub            stub.Stub
+	mask            stub.EventMask
+	options         Options // server options
+	runPodSandbox   func(*NriServer, *api.PodSandbox, *api.Container) error
```

Comment:
is it necessary to define these functions as member

Reply:
yes, it's unnessary, let's remove these codes.